### PR TITLE
doc: release: 2.4: summarize I2C changes

### PR DESCRIPTION
**Pull request by pabigot**
One new driver, one new capability.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

### CODE REVIEW

File: doc/releases/release-notes-2.4.rst [hunk 1]
```diff
@@ -229,6 +229,13 @@ Drivers and Sensors
 
 * I2C
 
+  * Introduced new driver for NXP LPC11U6x SoCs.
```
Comment (mbolivar-nordic) [inline doc/releases/release-notes-2.4.rst:232]:
```suggestion
  * Introduced new driver for NXP LPC11U6x SoCs. See :option:`CONFIG_I2C_LPC11U6X`.
```

File: doc/releases/release-notes-2.4.rst [hunk 2]
```diff
@@ -229,6 +229,13 @@ Drivers and Sensors
...
+
+  * Introduced new driver for emulated I2C devices, where I2C operations
+    are forwarded to a module that emulates responses from hardware.
+    This enables testing without hardware and allows unusual conditions
+    to be synthesized to test driver behavior.
```
Comment (mbolivar-nordic) [inline doc/releases/release-notes-2.4.rst:237]:
```suggestion
    to be synthesized to test driver behavior. See :option:`CONFIG_I2C_EMUL`.
```

**Review by mbolivar-nordic — COMMENTED** [review]
In the interest of not making the user grep, here are some suggested improvements.

(I see people asking questions on Slack about the existence of drivers that are easily solved with grep, so I assume users must not know how. Might as well make it easier on them ...)

File: doc/releases/release-notes-2.4.rst (hunk 2 above)
Comment (pabigot) [reply doc/releases/release-notes-2.4.rst]: Done (both).

**Review by mbolivar-nordic — APPROVED** [review]

**Review by sjg20 — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**